contracts/bridge/BridgeOperator.sol

```diff
@@ -51,29 +56,40 @@ contract BridgeOperator is Ownable {
     }
 
     // voteCommon handles common functionality for voting.
-    function voteCommon(VoteType voteType, bytes32 _voteKey)
+    function voteCommon(VoteType _voteType, uint64 _nonce, bytes32 _voteKey)
         internal
         returns(bool)
     {
-        if (!votes[_voteKey][msg.sender]) {
-            votes[_voteKey][msg.sender] = true;
-            require(votesCounts[_voteKey] < votesCounts[_voteKey] + 1, "votesCounts overflow");
-            votesCounts[_voteKey]++;
+        VotesData storage vote = votes[uint8(_voteType)][_nonce];
+
+        if (!vote.voted[_voteKey][msg.sender]) {
+            vote.voted[_voteKey][msg.sender] = true;
+
+            require(vote.voteCounts[_voteKey] < vote.voteCounts[_voteKey] + 1, "voteCounts overflow");
+            vote.voteCounts[_voteKey]++;
         }
-        if (votesCounts[_voteKey] >= operatorThresholds[uint8(voteType)]) {
+        if (vote.voteCounts[_voteKey] >= operatorThresholds[uint8(_voteType)]) {
             return true;
         }
         return false;
     }
 
+    // removeVoteData removes a vote data according to voteType and nonce.
+    function removeVoteData(VoteType _voteType, uint64 _nonce)
+        internal
+    {
+        delete votes[uint8(_voteType)][_nonce];
+    }
+
     // voteValueTransfer votes value transfer transaction with the operator.
     function voteValueTransfer(uint64 _requestNonce)
         internal
         returns(bool)
     {
         require(!closedValueTransferVotes[_requestNonce], "closed vote");
 
-        if (voteCommon(VoteType.ValueTransfer, keccak256(msg.data))) {
+        bytes32 voteKey = keccak256(msg.data);
+        if (voteCommon(VoteType.ValueTransfer, _requestNonce, voteKey)) {
             closedValueTransferVotes[_requestNonce] = true;
             return true;
```

Comment:
Do you have any reason not to call removeVoteData here?

Reply:
Good. As you said, `closedValueTransferVotes` can prevent to re-vote. So `removeVoteData` can be moved here.